src/coin-gecko/api.ts

```diff
@@ -16,74 +17,73 @@ type GetRosePriceParams = {
+  return useQuery<AxiosResponse<TokenPriceMap>, AxiosError<unknown>>(
+    ['tokenFiatPrices'],
     () =>
-      getRosePrice({
-        ids: 'oasis-network',
-        vs_currencies: 'usd',
+      getTokenPricesFromGecko({
+        ids: tokenIds.join(','),
+        vs_currencies: fiatCurrency ?? 'usd',
       }),
     {
-      select: ({ data }) => data['oasis-network'].usd,
+      select: ({ data }) => {
+        const result: TokenPriceMap = {}
+        Object.keys(data).forEach(key => {
+          result[key] = (data as any)[key][fiatCurrency]
+        })
+        return result as any
+      },
       staleTime,
     },
   )
 }
 
 export type TokenPriceInfo = {
   price?: number
+  fiatCurrency?: string
   isLoading: boolean
   isFree: boolean
   hasUsedCoinGecko: boolean
 }
 
-export const useTokenPrice = (ticker: NativeTicker): TokenPriceInfo => {
-  const { isLoading: roseIsLoading, data: rosePrice } = useGetRosePrice()
-  switch (ticker) {
-    case Ticker.ROSE:
-      return {
-        price: rosePrice,
-        isLoading: roseIsLoading,
-        isFree: false,
-        hasUsedCoinGecko: true,
-      }
-    case Ticker.TEST:
-      return {
-        hasUsedCoinGecko: false,
-        isLoading: false,
-        isFree: true,
-      }
-    default:
-      exhaustedTypeWarning('Checking price of unknown token', ticker)
-      return {
-        isLoading: false,
-        hasUsedCoinGecko: false,
-        isFree: false,
-      }
+export const useTokenPrice = (ticker: Ticker, fiatCurrency: string): TokenPriceInfo => {
+  const tokenPrices = useAllTokenPrices(fiatCurrency)
+  const price = tokenPrices[ticker]
+  if (!price) {
+    exhaustedTypeWarning('Checking price of unknown token ticker', ticker as any)
   }
+  return price || tokenPrices[Ticker.TEST]!
 }
 
-export type AllTokenPrices = Record<Network, TokenPriceInfo>
+export type AllTokenPrices = Partial<Record<Ticker, TokenPriceInfo>>
 
-export const useAllTokenPrices = (): AllTokenPrices => {
-  const results = {} as any
-  // The list of networks will never change on the run, so we can do this
-  // eslint-disable-next-line react-hooks/rules-of-hooks
-  RouteUtils.getEnabledNetworks().forEach(net => (results[net] = useTokenPrice(getTickerForNetwork(net))))
+export const useAllTokenPrices = (fiatCurrency: string): AllTokenPrices => {
+  const tokens = uniq(RouteUtils.getEnabledScopes().map(getTokensForScope).flat())
+  const geckoIds = tokens.map(token => token.geckoId).filter((id): id is string => !!id)
+  const { isLoading: geckoIsLoading, data: geckoPrices } = useGetTokenPricesFromGecko(geckoIds, fiatCurrency)
+  const results: AllTokenPrices = {}
+  tokens.forEach(token => {
+    results[token.ticker] = {
+      isLoading: geckoIsLoading,
+      isFree: !!token.free,
+      hasUsedCoinGecko: !!token.geckoId,
+      price: token.geckoId && geckoPrices ? (geckoPrices as any)[token.geckoId] : undefined,
+      fiatCurrency: token.geckoId && geckoPrices ? fiatCurrency : 'xx',
```

Comment:
'xx'?

Reply:
On second thought, we could simply use the provided  `fiatCurrency`, too. It doesn't matter, since (missing the price) it won't be displayed.

Reply:
Fixed in #1985.

Reply:
Fixed in #1285.